src/engine/sequencer.js

```diff
@@ -138,7 +138,6 @@ class Sequencer {
                     activeThread.status === Thread.STATUS_DONE) {
                     // Finished with this thread.
                     stoppedThread = true;
-                    this.runtime.updateCurrentMSecs();
```

Comment:
As discussed offline, I'm not sure I understand this change or whether it's necessary? It seems like it might be analogous to this code in Scratch 2, but I'm not sure:

https://github.com/LLK/scratch-flash/blob/develop/src/interpreter/Interpreter.as#L277

Reply:
Yup, I think this update of `currentMSecs` is analogous to that line in Scratch 2, or at least that's how I understood it when I added it initially. However after taking another look at it and some feedback from @mzgoddard, I've realized that we don't need this `currentMSecs` update because in Scratch 2 it was used to eventually end this while loop: https://github.com/LLK/scratch-flash/blob/develop/src/interpreter/Interpreter.as#L251 

And in Scratch 3, the analogous while loop uses the Timer util, which is updated by Date.now:

https://github.com/LLK/scratch-vm/blob/develop/src/engine/sequencer.js#L86-L89

I don't think this `currentMSecs` update is causing any bugs and I don't think removing it fixes any bugs, so I'll take it out of this PR and make a follow-up PR to research whether or not we want to take out that line.

Reply:
@ktbee and I discussed this change offline and it looks like the removal of this line is crucial to the bug this PR is fixing. Since we're now executing hats earlier (in order to catch the timer value at 0 instead of at some value > 0), we need to remove this update to currentMSecs right after the thread is killed.